packages/flutter/lib/src/material/banner_theme.dart

```diff
@@ -0,0 +1,135 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'theme.dart';
+
+/// Defines the visual properties of [MaterialBanner] widgets.
+///
+/// Descendant widgets obtain the current [MaterialBannerThemeData] object using
+/// `MaterialBannerTheme.of(context)`. Instances of [MaterialBannerThemeData]
+/// can be customized with [MaterialBannerThemeData.copyWith].
+///
+/// Typically a [MaterialBannerThemeData] is specified as part of the overall
+/// [Theme] with [ThemeData.bannerTheme].
+///
+/// All [MaterialBannerThemeData] properties are `null` by default. When null,
+/// the [MaterialBanner] will provide its own defaults.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+class MaterialBannerThemeData extends Diagnosticable {
+
+  /// Creates a theme that can be used for [ThemeData.bannerTheme].
+  const MaterialBannerThemeData({
+    this.backgroundColor,
+    this.contentTextStyle,
+  });
+
+  /// Default value for [MaterialBanner.backgroundColor].
+  ///
+  /// If null, [MaterialBanner] uses [ThemeData.canvasColor].
```

Comment:
For these, you should probably either omit the "if null" bit, or specify that if null, the `MaterialBanner` widget will use `ThemeData.canvasColor` by default.

Reply:
done